### PR TITLE
Add generateLayoutUnique

### PR DESCRIPTION
**Pull request by vdumont**
I'd like to propose adding `generateLayoutUnique`. This is something we had to implement on our end and could maybe benefit others.

**TL:DR;** if A.svg and B.svg are identical, a single icon will be in the sprite image and both A and B will reference the same image in the JSON.

Our use case is we have a data model with rich, granular shield types -- we may have hundreds of shields types, but on the map you may not have or want 1:1 artwork, you may have just templates (circles/rectangles/hexagons..) that you paint a certain color, similar to what Google Maps does.

We wanted to avoid having code to dumb down/map the granular data model shield types into simpler types that can map to icons directly for 2 reasons.

1) introducing new shield types shouldn't involve a code change, just the cartographic configuration and redeploying assets. 
2) with templates we can have the very basic (circle, rectangle, hexagon, custom, etc..) assets and dynamically generate more assets that cartographers don't have to maintain.

So we have something that looks like this:
```
-------------------------------------------------------------
shield_type     |       template            |   paint_color
-------------------------------------------------------------
USA_FLORIDA_STATE      circle.svg             white
USA_CALIFORNIA_STATE   circle.svg             green
USA_MAINE_STATE        rectangle.svg          yellow
```
 
(and potentially hundreds more).

With this information we can generate USA_FLORIDA_STATE.svg that is circle.svg painted white, and USA_CALIFORNIA_STATE.svg that is circle.svg painted green and so on...

We can also have different configurations for different stylesheets, potentially using the same templates.

Then we can just have a single style rule per shield text color, which is a big plus. So we avoid having multiple style rules for different shield types. This works really well.

The only downside is that if 2 assets have (or temporarily have) the same template and paint_color, their resulting SVG looks identical but with different names. Imagine we'd add USA_OREGON_STATE that is also circle.svg and green paint color. USA_CALIFORNIA_STATE.svg and USA_OREGON_STATE.svg would look identical and will be packed twice in the sprite png, since spritezero is based on file names. This wastes space (imagine all US states would all be circle.svg and green). 

Since the stylesheet rule look up the JSON with the {shield} to know what icon to fetch, the JSON entry  for both still needs to be preserved though.

This is what this patch does, it dedupes SVGs so that identical SVGs are packed only once per sprite, but their reference in the JSON is preserved. 

Happy to elaborate if this doesn't make much sense :)

### CODE REVIEW

**Comment by vicapow** [comment]
+@jfirebaugh, +@bhousel sorry to ping you but could you take a look? For our use cases, we've notice a 45% reduction in the final sprite size.

File: index.js [hunk 1]
```diff
@@ -43,20 +91,27 @@ function generateLayout(imgs, pixelRatio, format, callback) {
         if (err) return callback(err);
 
         imagesWithSizes.sort(heightAscThanNameComparator);
-
+
```
Comment (tmcw) [inline index.js]: please remove the space-only line here

**Comment by vdumont** [comment]
@tmcw sorry to pester, since you have the most commits in this repo, any thoughts on this? I should also point out that everything is backward compatible, this is essentially adding a function for the --unique functionality.

**Comment by vicapow** [comment]
Following up again. Is there interest in merging this?

**Comment by bhousel** [comment]
Hey @vicapow, sorry for the slowness, backed up with a lot of other priorities..
Yes I'll probably have time to merge this in the next few days.

**Comment by vdumont** [comment]
@bhousel Thanks Bryan!
